Update TURN.md

MichaIng opened the pull request:
+ In `turnserver.conf`, `lt-cred-mech` is obsolete/conflicting in combination with `use-auth-secret`

  Related log entries:

      Jan 19 23:08:20 my.domain.org turnserver[6712]: CONFIGURATION ALERT: You specified --lt-cred-mech and --use-auth-secret in the same time.
      Jan 19 23:08:20 my.domain.org turnserver[6712]: Be aware that you could not mix the username/password and the shared secret based auth methohds.
      Jan 19 23:08:20 my.domain.org turnserver[6712]: Shared secret overrides username/password based auth method. Check your configuration!

  Related config file comments:

      # Be aware that use-auth-secret overrides some part of lt-cred-mech.
      # Notice that this feature depends internally on lt-cred-mech, so if you set
      # use-auth-secret then it enables internally automatically lt-cred-mech option
      # like if you enable both.
      #
      # You can use only one of the to auth mechanisms in the same time because,
      # both mechanism use the username and password validation in different way.
      #
      # This way be aware that you can't use both auth mechnaism in the same time!
      # Use in config either the lt-cred-mech or the use-auth-secret
      # to avoid any confusion.

  - [x] Successfully tested TURN server functionality with Nextcloud Talk desktop browser <> Android app connection through NAT, without `lt-cred-mech`
+ Info: On Debian Buster, respectively the related `coturn` version, a systemd unit has been implemented, superseding the init.d service.
+ Minor visual and wording changes
+ Keep in sync with: https://help.nextcloud.com/t/howto-setup-nextcloud-talk-with-turn-server/30794

**Signed-off-by: Micha Felle <micha@dietpi.com>**

mario commented:
Thank you very much! :)

wiswedel commented:
@mario why is this in the Android repo and not in https://github.com/nextcloud/spreed/?

mario commented:
@wiswedel the goal was to move it to https://docs.nextcloud.com, but got sidetracked.

@MichaIng fancy submitting another PR? :)

https://github.com/nextcloud/documentation

wiswedel commented:
@mario https://docs.nextcloud.com (https://github.com/nextcloud/documentation) only documents the server and the built-in-shipped apps.
The TURN docs are better off under https://github.com/nextcloud/spreed.

@MichaIng If you don't mind, please submit your PR for https://github.com/nextcloud/spreed/tree/master/docs.

MichaIng commented:
@wiswedel @mario 
Jep will do that. Requires update anyway to keep in sync with the HowTo in forum, e.g. info that TLS is unnecessary since currently not supported by Nextcloud Talk. Btw docs also include some info about non-builtin apps, e.g. calendar and contacts: https://github.com/nextcloud/documentation/tree/master/user_manual/pim
However indeed it fits better to the Talk server repo than Talk client repo, as it needs to be configured server side.

MichaIng commented:
@wiswedel @mario 
Done: https://github.com/nextcloud/spreed/pull/2142
- Updated to match current HowTo from forum: https://help.nextcloud.com/t/howto-setup-nextcloud-talk-with-turn-server/30794
- Slight update based on current knowledge and slight rework, so that it reads more like a professional documentation, rather then a forum thread.
- Fixed spelling.
- Fixed format for GitHub markdown viewer.

wiswedel commented:
👍 Thanks, @MichaIng
